server/daos/lecture_dao.js

```diff
@@ -0,0 +1,52 @@
+// import database
+// import modules
+
+const db = require('../db/db.js');
+const Lecture = require('../models/lecture.js');
+
+const createLecture = function (row){
+    return new Course(row.id,row.datetime,row.course_id,row.room_id,row.virtual,row.deleted_at);
+}
+
+// it creates the lecture table
+exports.createLectureTable = function() {
+    return new Promise ((resolve,reject) => {
+        const sql = `CREATE TABLE Lectures (id INTEGER NOT NULL PRIMARY KEY, datetime TEXT, course_id TEXT NOT NULL,
+                     room_id TEXT NOT NULL, virtual BOOLEAN NOT NULL DEFAULT (0) CHECK (virtual IN (0,1)),
+                     deleted_at TEXT, FOREIGN KEY(course_id) REFERENCES Courses(id), FOREIGN KEY(room_id) REFERENCES Rooms(id))`
```

Comment:
deleted_at should be timestamp

Reply:
I replied in a previous comment

Reply:
which it disappears.. I told you that reading sqlite doc varchar and timestamp are not native types but they are approximated with text.
Here is what I found:
https://www.sqlite.org/datatype3.html

Reply:
You are right, sorry 😅